source/lib/course-search/load-cached-courses.js

```diff
@@ -0,0 +1,26 @@
+// @flow
+
+import type {CourseType, TermType} from './types'
+import {loadTermsFromStorage} from './update-course-storage'
+import {COURSE_STORAGE_DIR} from './urls'
+import RNFS from 'react-native-fs'
+
+export async function loadCachedCourses(): Array<CourseType> {
+	const terms: Array<TermType> = await loadTermsFromStorage()
+	let courses: Array<CourseType> = []
+
+	for (let i = 0; i < terms.length; i++) {
+		let newCourses: Array<CourseType> = await loadTermCoursesFromStorage(
+			terms[i],
+		)
+		courses.push(...newCourses)
+	}
+	// console.log(courses)
+	return courses
```

Comment:
To appease the `eslint` gods:

```js
import flatten from 'lodash/flatten'
const promises = terms.map(term => loadTermCoursesFromStorage(term))
const coursesByTerm = await Promise.all(promises)
return flatten(coursesByTerm)
```

To understand the `eslint` gods: If you have an array and loop over it, `await`ing in each iteration, you're resolving the promises sequentially. If you can rework the iteration to use `Promise.all`, then the promises can resolve in parallel.

Does that code snippet make sense? I can explain what it does in more detail.